src/android/NotificationHandler.java

```diff
@@ -0,0 +1,115 @@
+package com.spoon.backgroundfileupload;
+
+import android.app.Activity;
+import android.app.NotificationManager;
+import android.content.res.Resources;
+import android.widget.RemoteViews;
+
+import androidx.core.app.NotificationCompat;
+
+import net.gotev.uploadservice.UploadService;
+import net.gotev.uploadservice.observer.task.AbstractSingleNotificationHandler;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+public class NotificationHandler extends AbstractSingleNotificationHandler {
+
+    private Activity mContext;
+
+    public NotificationHandler(@NotNull UploadService service, Activity context) {
+        super(service);
+        this.mContext = context;
+    }
+
+    @Nullable
+    @Override
+    public NotificationCompat.Builder updateNotification(@NotNull NotificationManager notificationManager, @NotNull NotificationCompat.Builder builder, @NotNull Map<String, TaskData> map) {
+        int speed = 0;
+        int inProgress = 0;
+
+        for (Map.Entry<String, TaskData> entry : map.entrySet()) {
+            String uploadId = entry.getValue().getInfo().getUploadId();
+
+            if (entry.getValue().getStatus() == TaskStatus.InProgress) {
+                inProgress++;
+
+                speed += convertUnit(
+                        entry.getValue().getInfo().getUploadRate().getUnit().name(),
+                        entry.getValue().getInfo().getUploadRate().getValue()
+                );
+            }
+
+            if (entry.getValue().getStatus() == TaskStatus.Failed || entry.getValue().getStatus() == TaskStatus.Succeeded) {
+                removeTask(uploadId);
+            }
+        }
+
+        String pkg = mContext.getApplication().getPackageName();
+        String layoutDef = "layout";
+        String idDef = "id";
+
+        Resources resources = mContext.getResources();
+
+        RemoteViews notificationLayout = new RemoteViews(mContext.getPackageName(),
+                resources.getIdentifier("notification_small", layoutDef, pkg));
+
+        notificationLayout.setTextViewText(
+                resources.getIdentifier("notification_title", idDef, pkg),
+                String.format("%s uploads remaining", PendingUpload.all().size())
```

Comment:
```suggestion
                String.format("%s uploads remaining", uploading)
```

Et faire le décompte uploading directement dans la boucle sur les taches (pour éviter de faire une requete SQL dans l'update des notifications !